### PR TITLE
answer coalesce via enrichment

### PR DESCRIPTION
**Pull request by cbizon**
An operation for creating new results by combining input results via enrichment analyses.

### CODE REVIEW

File: operations/enrich_results.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+description: Create new results by applying enrichment analysis to existing results.  In particular, combines results by transforming a qnode into a set, formed of knodes that share a property or relation more often than expected by chance.
+unique: true
+examples:
+- input: enrich/messages/01_input_enrich_results.json
+  output: enrich/messages/02_output_enrich_results.json
+input_requirements: []
+output_guarantees: []
+allowed_changes:
+- add qnodes
+- add qnodes
```
Comment (patrickkwang) [inline operations/enrich_results.yml:12]:
```suggestion
- add qedges
```

File: operations/enrich_results.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+- modify qnodes
+- add kedges
```
Comment (patrickkwang) [inline operations/enrich_results.yml:14]:
```suggestion
- add knodes
```

File: operations/enrich_results.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      type: float
+      example: 1e-7 
+      minimum: 0
+      maximum: 1
+      default: 1e-6
+    qnode_keys:
+      description: If specified, then only knodes bound to these qnodes will be examined for enrichment and combination. 
+      type: array
+      example: [n01]
+      default: []
```
Comment (patrickkwang) [inline operations/enrich_results.yml:31]:
```suggestion
```
IMO the default `[]` here is counter-intuitive. The description implies what happens when this key is left out, so I don't think we need a default. We could even go as far as to insist that the array be non-empty with `minLength: 1`.

**Review by patrickkwang — CHANGES_REQUESTED** [review]
Makes sense! Minor suggestions.

File: docs/enrich_results.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+```yaml
+properties:
+  pvalue_threshold:
+    default: 1e-6
+    description: The cutoff p-value for enrichment.
+    example: 1e-7
+    maximum: 1
+    minimum: 0
+    type: float
```
Comment (dkoslicki) [inline docs/enrich_results.md:36]: type: number?

File: docs/enrich_results.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  qnode_keys:
+    default: []
+    description: If specified, then only knodes bound to these qnodes will be examined
+      for enrichment and combination.
+    example:
+    - n01
+    type: array
```
Comment (dkoslicki) [inline docs/enrich_results.md:43]: item types?

**Review by dkoslicki — CHANGES_REQUESTED** [review]

File: docs/enrich_results.md (hunk 4 above)
Comment (cbizon) [reply docs/enrich_results.md:36]: Changed to float

File: docs/enrich_results.md (hunk 5 above)
Comment (cbizon) [reply docs/enrich_results.md:43]: added item type strings

**Review by patrickkwang — APPROVED** [review]

**Comment by dkoslicki** [comment]
@cbizon I think the YAML was changed, but the docs/markdown files haven't been recompiled

**Review by dkoslicki — APPROVED** [review]

**Pull request merged**